webapp/channels/src/actions/notification_actions.jsx

```diff
@@ -49,19 +49,19 @@ const getNotificationSoundFromChannelMemberAndUser = (member, user) => {
 };
 
 /**
- * @returns {import('mattermost-redux/types/actions').ThunkActionFunc<void>}
+ * @returns {import('mattermost-redux/types/actions').ThunkActionFunc<Promise<NotificationResult>, GlobalState>}
  */
 export function sendDesktopNotification(post, msgProps) {
     return async (dispatch, getState) => {
         const state = getState();
         const currentUserId = getCurrentUserId(state);
 
         if ((currentUserId === post.user_id && post.props.from_webhook !== 'true')) {
-            return;
+            return {result: 'not_sent', reason: 'own_post'};
```

Comment:
Not something that needs to be changed at this point since the code is already messy, but this has the same problem as the serverside notification code that it interleaves logic for so many things at once. In this case, it's mixing the logic for determining whether or not to show the notification with constructing the notification which makes it hard to follow

Reply:
Yes, I would think this close and the server code should be cleaned up as a follow-up to this work.